Mathlib/Algebra/Order/Group/Basic.lean

```diff
@@ -137,3 +137,77 @@ lemma zpow_eq_zpow_iff' (hn : n ≠ 0) : a ^ n = b ^ n ↔ a = b := zpow_left_in
 #align zsmul_eq_zsmul_iff' zsmul_eq_zsmul_iff'
 
 end LinearOrderedCommGroup
+
+section LinearOrderedAddCommGroupWithTop
+
+variable (α : Type*) [LinearOrderedAddCommGroupWithTop α]
+
+@[simp]
+lemma top_ne_zero' :
+    (⊤ : α) ≠ 0 := by
+  intro nh
+  have ⟨a, b, h⟩ := Nontrivial.exists_pair_ne (α := α)
+  have : a + 0 ≠ b + 0 := by simpa
+  rw [← nh] at this
+  simp at this
+
+@[simp]
+lemma neg_eq_top_iff (a : α) :
+    -a = ⊤ ↔ a = ⊤ :=
+  ⟨by
+  intro h
+  by_contra nh
+  replace nh := add_neg_cancel_of_ne_top nh
+  rw [h, add_top] at nh
+  exact top_ne_zero' _ nh, fun h ↦ h ▸ neg_top⟩
+
+@[simp]
+lemma add_eq_top_iff (a b : α) :
+    (a + b = ⊤) ↔ a = ⊤ ∨ b = ⊤ := by
```

Comment:
```suggestion
lemma add_eq_top {a b : α} :
    a + b = ⊤ ↔ a = ⊤ ∨ b = ⊤ := by
```
```suggestion
lemma add_eq_top_iff (a b : α) :
    (a + b = ⊤) ↔ a = ⊤ ∨ b = ⊤ := by
```

Reply:
This collides with `ENNReal.add_eq_top` so I had to undo this